Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerViewController.swift

```diff
@@ -279,16 +282,34 @@ open class MessageComposerViewController<ExtraData: ExtraDataTypes>: ViewControl
     }
     
     // MARK: Suggestions
-    
-    func showSuggestionsViewController(
-        with state: SuggestionsViewControllerState,
-        onSelectItem: ((Int) -> Void)
-    ) {
-        suggestionsPresenter?.showSuggestionsViewController(with: state, onSelectItem: onSelectItem)
+
+    public func showSuggestionsViewController(for kind: SuggestionKind, onSelectItem: @escaping ((Int) -> Void)) {
+        guard let parent = parent else { return }
+
+        let dataSource: UICollectionViewDataSource
+        switch kind {
+        case let .command(hints):
+            dataSource = SuggestionsCommandDataSource(
+                with: hints,
+                collectionView: suggestionsViewController.collectionView
+            )
+        case .mention:
+            dataSource = SuggestionsMentionDataSource(
+                collectionView: suggestionsViewController.collectionView,
+                searchController: userSuggestionSearchController
+            )
+        }
+        suggestionsViewController.didSelectItemAt = onSelectItem
+        suggestionsViewController.dataSource = dataSource
+        suggestionsViewController.updateContentIfNeeded()
```

Comment:
Why do you call this form the outside? The VC presented VC should handle this internally.